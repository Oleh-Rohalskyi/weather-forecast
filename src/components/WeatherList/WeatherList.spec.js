import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import WheatherList from './WeatherList';

describe('src > components > WeatherList', () => {

  const mockUp = [{
    clouds: { all: 100 },
    dt: 1581768000,
    dt_txt: "2020-02-15 12:00:00",
    main: { temp: 283.47, feels_like: 278.81},
    sys: { pod: "d" },
    weather: [{ id: 804, main: "Clouds", description: "overcast clouds", icon: "04d" }],
    wind: { speed: 5.19, deg: 199 },
  },{
    clouds: {all: 95},
    dt: 1581778800,
    dt_txt: "2020-02-15 15:00:00",
    main: {temp: 283.93, feels_like: 279.39},
    sys: {pod: "d"},
    weather: [{id: 804, main: "Clouds", description: "overcast clouds", icon: "04d"}],
    wind: {speed: 5.1, deg: 195},
  }]


  const mockStore = configureStore({
    weather: {
      list: []
    }
  });
  const store = mockStore({
    weather: {
      list: mockUp
    }
  });

  jest.spyOn(store, 'dispatch');

  beforeEach(() => {
    store.dispatch.mockClear();
  });

  it('renders without crashing', () => {
    
    const wrapper = mount(
      <Provider store={store}>
        <WheatherList />
      </Provider>
    );

    expect(wrapper).toMatchSnapshot();

  });

});
