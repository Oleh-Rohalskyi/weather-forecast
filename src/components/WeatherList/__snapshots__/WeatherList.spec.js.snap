// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src > components > WeatherList renders without crashing 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <WeatherList>
    <ul
      className="weather-list"
    >
      <WeatherItem
        key="0"
        props={
          Object {
            "description": "overcast clouds",
            "dt_txt": "2020-02-15 12:00:00",
            "feels_like": 278.81,
            "temp": 283.47,
          }
        }
      >
        <li
          className="weather-item"
        >
          <div
            className="weather-item__info"
          >
            <span>
              2020-02-15
            </span>
            <span>
              12:00:00
            </span>
          </div>
          <div
            className="weather-item__data"
          >
            <p>
              temperature: 
              283.47
            </p>
            <p>
              feels like temperature: 
              278.81
            </p>
          </div>
          <div
            className="weather-item__chip"
          >
            overcast clouds
          </div>
        </li>
      </WeatherItem>
      <WeatherItem
        key="1"
        props={
          Object {
            "description": "overcast clouds",
            "dt_txt": "2020-02-15 15:00:00",
            "feels_like": 279.39,
            "temp": 283.93,
          }
        }
      >
        <li
          className="weather-item"
        >
          <div
            className="weather-item__info"
          >
            <span>
              2020-02-15
            </span>
            <span>
              15:00:00
            </span>
          </div>
          <div
            className="weather-item__data"
          >
            <p>
              temperature: 
              283.93
            </p>
            <p>
              feels like temperature: 
              279.39
            </p>
          </div>
          <div
            className="weather-item__chip"
          >
            overcast clouds
          </div>
        </li>
      </WeatherItem>
    </ul>
  </WeatherList>
</Provider>
`;
